HFSSdrawpy/core/port.py

```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
             posx = port.pos[0]*np.cos(rad)+port.pos[1]*np.sin(-rad)
             posy = port.pos[0]*np.sin(rad)+port.pos[1]*np.cos(rad)
             port.pos = Vector([posx, posy])
+
+    def split(self, subnames=[], gap=None):
+        """
+        Creates CPW like ports out of the listed subports
+
+        Parameters
+        ----------
+        subnames : list of str, must be part of eponymous instance attribute
+                   if empty, split all ports
+                   if -1 is passed, split all ports but last
+        gap      : str, gap size for the output CPW ports
+                   if None, track size is used
+        Returns
+        -------
+        subports : list of Port instances
+
+        """
+
+        if not subnames:
+            subnames = self.subnames
+        elif subnames == -1:
+            subnames = self.subnames[:-1]
+        if not all(sub in self.subnames for sub in subnames):
+            raise ValueError('Split ports must belong to '+str(self.subnames))
+
+        self.save = {'name':self.name, 'pos':self.pos, 'widths':self.widths,
+                     'subnames':self.subnames, 'offsets':self.offsets}
```

Comment:
This is not needed here (I think). The save is only required when performing a brief modification of the port (such as when drawing adaptors, the port is temporarily shifted and resized to the other port specs).